### PR TITLE
Remove component from PR title in rake posts:new

### PR DESCRIPTION
**Pull request by jonatack**
A few force pushes to simplify the change and add comments.

Added logging to see the results while running `rake posts:new`.

Handle some edge cases seen while testing.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -29,6 +29,8 @@ DESIRED_COMPONENTS = [
   'Validation',
   'Wallet',
 ].freeze
+COMPONENTS = DESIRED_COMPONENTS.map(&:downcase).freeze
+UNDESIRED_PR_TITLE_WORDS = %w(- _).freeze
```
Comment (jnewbery) [inline Rakefile]: Can you comment what this is matching on/doing?

File: Rakefile [hunk 2]
```diff
@@ -159,9 +169,32 @@ def create_post_file!(filename, response, date, host)
 end
 
 def parse_title(title)
-  "\"#{title.gsub(/"/,  '')}\""
+  first, *rest = title.split # e.g. if title = "a b c", first = "a", rest = ["b", "c"]
+  first.downcase! # mutate first word to lowercase in place
+  rest.shift if UNDESIRED_PR_TITLE_WORDS.include?(rest[0]) # rm 1st word if indesired
```
Comment (jnewbery) [inline Rakefile]: s/indesired/undesired

File: Rakefile [hunk 3]
```diff
@@ -171,13 +171,17 @@ end
...
-  # If prefix is different from first word and is a component, set title to the rest.
-  title = rest.join(' ') if first != prefix && is_a_component?(prefix)
-
-  # Return title enclosed in double quotes after removing any double quotes.
-  "\"#{title.gsub(/"/,  '')}\""
+  # If prefix is different from first word and is a component, drop first word.
+  words = if first != prefix && is_a_component?(prefix)
+            [rest.first&.capitalize] + rest[1..-1]
+          else
+            [first&.capitalize!] + rest
```
Comment (jonatack) [inline Rakefile]: Re-reading, I think `capitalize!` is a bug here and should be `capitalize`.

File: Rakefile (hunk 1 above)
Comment (jonatack) [reply Rakefile]: Sure. While testing different PRs after adding the logging, I came across ones that began with "- (rest of title)" or "_ (rest of title)" after the prefix was removed.

Any unwanted words at the start of the post PR title like that (in this case, single character words) can be added to this array to be removed.

`%w(- _)`

is array shorthand for

`["-", "_"]`

File: Rakefile (hunk 2 above)
Comment (jonatack) [reply Rakefile]: Thanks! That's French slipping into my English :D

File: Rakefile (hunk 1 above)
Comment (jnewbery) [reply Rakefile]: Thanks. Perhaps you could add a comment, including example PR titles that get fixed by this.

**Comment by jonatack** [comment]
Thanks for reviewing. Rebased after fixing the code comment and my comment (I don't think it was a bug but it was not consistent) and did more testing. The added logging does make it easy to see if there is any unexpected behavior.

File: Rakefile (hunk 1 above)
Comment (jonatack) [reply Rakefile]: Good idea. Here's one:
```
rake -- posts:new -h a -p 16729
Date set to next Wednesday: 2019-12-25
GitHub PR title:  "[GUI] - Reintroduce menu icons"
Parsed PR title:  "Reintroduce menu icons"
GitHub PR labels: ""
Parsed PR labels: ""
New file _posts/2019-12-25-#16729.md created successfully.
```

Comment (jonatack) [reply Rakefile]: Done

**Comment by jnewbery** [comment]
Looks good. Thanks Jon!

**Pull request merged**